cpp/src/arrow/flight/test-integration-client.cc

```diff
@@ -49,6 +48,38 @@ int main(int argc, char** argv) {
 
   arrow::flight::FlightDescriptor descr{
       arrow::flight::FlightDescriptor::PATH, "", {FLAGS_path}};
+
+  // 1. Put the data to the server.
+  std::unique_ptr<arrow::ipc::internal::json::JsonReader> reader;
+  std::shared_ptr<arrow::io::ReadableFile> in_file;
+  std::cout << "Opening JSON file '" << FLAGS_path << "'" << std::endl;
+  ABORT_NOT_OK(arrow::io::ReadableFile::Open(FLAGS_path, &in_file));
+
+  int64_t file_size = 0;
+  ABORT_NOT_OK(in_file->GetSize(&file_size));
+
+  std::shared_ptr<arrow::Buffer> json_buffer;
+  ABORT_NOT_OK(in_file->Read(file_size, &json_buffer));
+
+  ABORT_NOT_OK(arrow::ipc::internal::json::JsonReader::Open(json_buffer, &reader));
```

Comment:
Factor lines 54 through 64 into a method, then maybe we can add it to arrow/ipc/json.h? see https://issues.apache.org/jira/browse/ARROW-4409